esiclient/tests/functional/utils.py

```diff
@@ -0,0 +1,68 @@
+#    Copyright (c) 2016 Mirantis, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import json
+
+from tempest.lib.common.utils import data_utils
+from tempest.lib import exceptions
+
+
+def node_create(client, driver='fake-hardware', name=None, params='',
+                fail_ok=False, merge_stderr=False):
+
+    if not name:
+        name = data_utils.rand_name('baremetal')
+
+    output = client.openstack('baremetal node create '
```

Comment:
Just so things are a bit cleaner, a `baremetal(...)` method should probably be added to the CLIClient class in base.py. This would make this call and calls like it a bit cleaner and would be a bit more consistent with the way it's done in the python-esileapclient repo (`client.openstack('baremetal node create', ...)` becomes `client.baremetal('node create', ...)`, etc.). It would also simplify the passing of positional arguments since if you use the client.openstack method you basically have to pass everything in as parameters which is a bit annoying sometimes.

Reply:
im going to push back and say for the sake of getting something merged now, this is ok. Poor Mainn has been working off this pr and I'd hate to pull the rug from underneath him. Perhaps this can be a fixed in a followup pr

Reply:
Ah, feel free to fix this now :) I was wondering what the purpose of the clients in CLIClient were! I think all you'd have to do is rename the ironic client in CLIClient to baremetal, and do a replace-all of "client.openstack('baremetal" to "client.baremetal('"

Reply:
I didnt realize you could do client.baremetal. But if I'm understanding correctly Sam is suggesting having a more general options parser like in https://github.com/CCI-MOC/python-esileapclient/blob/c18f51a49107f4f87af555fde4fdca614717ee8c/esileapclient/tests/functional/utils/output_utils.py#L1

Reply:
I can refactor it this way, but going through specific baremetal and the eventual neutron CLIClient objects seemed overly complex to just spelling out exactly what we need in one function